crates/biome_lsp/src/handlers/text_document.rs

```diff
@@ -20,17 +21,36 @@ use tracing::{error, field};
 pub(crate) async fn did_open(
     session: &Session,
     params: lsp_types::DidOpenTextDocumentParams,
-) -> Result<()> {
+) -> Result<(), LspError> {
     let url = params.text_document.uri;
     let version = params.text_document.version;
     let content = params.text_document.text;
     let language_hint = DocumentFileSource::from_language_id(&params.text_document.language_id);
 
-    let biome_path = session.file_path(&url)?;
-    let doc = Document::new(version, &content);
+    let path = session.file_path(&url)?;
+    let project_key = match session.project_for_path(&path) {
+        Some(project_key) => project_key,
+        None => {
+            info!("No open project for path: {path:?}. Opening new project.");
+            let parent_path = BiomePath::new(
+                path.parent()
+                    .map(|parent| parent.to_path_buf())
+                    .unwrap_or_default(),
+            );
+            let project_key = session.workspace.open_project(OpenProjectParams {
+                path: parent_path.clone(),
+                open_uninitialized: true,
```

Comment:
I am slightly afraid that setting this setting to `true` here may lead to the LSP opening a lot of projects whenever you open files in arbitrary directories that are not part of a Biome project. But setting to `false` currently breaks a lot of tests, so we may just need to see how this develops.